alternator/rjson.hh

```diff
@@ -146,6 +148,11 @@ const rjson::value* find(const rjson::value& value, std::string_view name);
 rjson::value& get(rjson::value& value, std::string_view name);
 const rjson::value& get(const rjson::value& value, std::string_view name);
 
+template<typename T>
```

Comment:
Yes. More to the point, to be able to marshall more or less complex objects automatically. Since the facilities are there for it. 
See stream_id, UUID, etc etc.

Reply:
Yes. I think we just have a disagreement in style. I also don't like ugly code, but I don't think that seeing a function call like to_value(stream_id) is ugly. I'm worried about code which is too "automatic" (use operator overloading and templates for everything) because it makes it almost impossible to understand how certain objects get written into the json and the IDE can't help you figure out which function is called to do the conversion. I'm also worried of the Alternator getting two completely different styles - some code uses the existing explict functions, some code uses your new implicit conversions - which will be hard to understand or maintain.
But let's address the style issue last I guess. I'm not married to the existing style.

Reply:
Again, our whole concept or serialization is based on templated functions choosing impl-helpers by type matching. It is a prevalent style/design in scylla and c++ in general. 
The general idea is that the typedef of the argument should not impact how you send it to the receiver. I.e. `put(A)` should look the same if A is a complex type/ref or  if it is an int. If you force `put(A.to_string())`, this is not the case (no `to_string` in `int`). Having free functions `to_string(...)` works, but then you are just moving overloading to a function call, basically costing you boilerplate typing and really gaining you no clarification as to which of the `to_string` functions you called (in the general case).